internal/resources/schemas/nsx_t_network_schemas.go

```diff
@@ -0,0 +1,82 @@
+package schemas
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	DSNetworkPool = "hpegl_vmaas_network_pool"
+	DSDhcpServer  = "hpegl_vmaas_dhcp_server"
+)
+
+func StaticNetworkSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeList,
+		Optional:    true,
+		Description: "static Network configuration",
+		MaxItems:    1,
+		ExactlyOneOf: []string{
+			"static_network",
+			"dhcp_network",
+		},
+		ConflictsWith: []string{
+			"dhcp_network",
+		},
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"pool_id": {
+					Type:        schema.TypeInt,
+					Optional:    true,
```

Comment:
when pool_id is optional, do you need to check if the data is empty inside this ? which you are doing in diff validate..

These changes may break the acceptance test cases i guess.

Reply:
updated